### ISSUE TITLE
Auto-Update DayZ mods

### ISSUE THREAD

**Issue by Osiris-Team** [Osiris-Team/AutoPlug-Client#215]
The Steam workshop id of a mod can be found in: `./@INSERT_MOD_NAME/meta.cpp` which looks something like this:
```props
protocol = 1;
publishedid = 1559212036;
name = "CF";
timestamp = 5249804932187309401;
```
This is basically a Java properties file (expect the semicolons, that would have to be removed when reading the values).

Currently, all mods are in the same folder as the server, which isn't really nice. A workaround for this would be to use the `-mod=ABOSULTE_PATH_MOD1;ABSOLUTE_PATH_MOD2;etc...` argument to individually point to each mod directory which does not work really well according to testing by users.

The workshop mods are originally located in `INSERT_DRIVE\SteamLibrary\steamapps\Common\DayZ\!Workshop`. The question is if these should be moved into the server's directory or only symlinks should be provided to the original folders / wanted mods.

**Comment by JustABiologist** [comment]
Submitted PR #280 for the 50€ bounty: https://github.com/Osiris-Team/AutoPlug-Client/pull/280

Payment request if accepted/merged: PayPal stab.me.papi@gmail.com

**Comment by jtc268** [comment]
Submitted PR #284 for the 50 EUR bounty: https://github.com/Osiris-Team/AutoPlug-Client/pull/284

Scope: generic Steam Workshop mod auto-update support via `meta.cpp` published IDs, wired through `ResourceFinder`, using the existing numeric `server-updater.software` Steam app id, SteamCMD workshop downloads, config caching, and NOTIFY/MANUAL/AUTOMATIC profile handling.

Validation:
- `mvn -q -DskipTests compile`
- `mvn -q -Dtest=SteamWorkshopModTest,SteamWorkshopSearchTest,TaskSteamWorkshopModDownloadTest,SteamCMDTest#buildsWorkshopDownloadCommand test`
- `git diff --check`

No payout details are posted publicly.

**Comment by keilogic** [comment]
Hi, if #280 is inactive and this 50 EUR bounty is still open to a fresh implementation, I can take a clean pass aligned with the existing review notes: generic Steam Workshop support using the configured Steam app ID, metadata caching in mods.yml, ResourceFinder-based update checks, SteamCMD workshop download integration, and focused tests. I will avoid duplicating #280 unless you confirm you want a new PR.

**Comment by Osiris-Team** [comment]
@keilogic sure, just make sure to make a picture with your phone from your PR and add it there.

**Comment by keilogic** [comment]
Thanks for confirming. One clarification before I start a fresh implementation: I can attach a regular screenshot plus full validation logs with the PR, but I do not have a phone-camera proof path available from this environment. Is a normal screenshot acceptable, or is a phone photo mandatory for review?

**Comment by Osiris-Team** [comment]
@keilogic to be honest that already sounds like an AI answer. Fully AI generated PRs are not allowed, the commiter must be a human and validate the PR first before creating it. Using AI as assistance is okay.

**Comment by Zoriot** [comment]
Is it fine to build upon #280, it seems that pr was quite far in the review. 
Would be nice if I can apply for the bounty then.'
Also i wonder if the condition ``Your PayPal email must be visible somewhere on your public GitHub profile before the PR is merged.`` is really required and if yes, what's the reasoning behind it? I don't feel save doing that rn

**Comment by Osiris-Team** [comment]
@Zoriot the reasoning is that I need to have a direct link between your github and paypal account, to avoid payments to an impersonator of you.

**Comment by Osiris-Team** [comment]
@Zoriot yes its fine to build upon those PRs, the bounty is awarded to the person creating the first mergable PR.

**Comment by Osiris-Team** [comment]
@Zoriot I updated the bounty terms to include more details.

**Comment by Zoriot** [comment]
Thanks, sounds good. I like the changes made it, the Fairness Grace Period also sounds reasonable, at the same time might have the risk that the person then just uses the code of the other pr. 

Regarding the email: Would it be fine when the email on GitHub contacts you with the PayPal details? With that you can still verify the connection and also allow more privacy.

**Comment by Osiris-Team** [comment]
@Zoriot is there a place on github where I can see your github email? If so yeah that works too.

**Comment by lymeow233-wq** [comment]
One technical scope check before I finish real validation: when a dedicated server App ID differs from the Workshop consumer App ID, which should be passed to `workshop_download_item`? For the issue's DayZ example, `server-updater.software` would normally be the dedicated-server App ID, while Steam's published-file-details response for item `1559212036` reports `consumer_app_id=221100`. Should the updater always reuse `server-updater.software`, or use the item's `consumer_app_id` for the Workshop download?

**Comment by Osiris-Team** [comment]
@lymeow233-wq great question! I'm not entirely sure about the difference between those IDs myself. Could you do some research on that? It sounds more like an implementation detail than a technical scope check, that we can discuss in the actual PR later.

**Comment by JnCaoxia** [comment]
Submitted PR #290 for the 50€ bounty: https://github.com/Osiris-Team/AutoPlug-Client/pull/290

**Comment by vernicular22** [comment]
Hi — I’m interested in taking this over from #290. Before I start, could you confirm that a derived PR based on #290 is still eligible under the current bounty terms?

For payout verification, may I send the PayPal details privately from the verified email attached to my GitHub account instead of publishing the PayPal address on my profile? I’m happy to prove control without exposing the payment address publicly.

I’ll review and test the implementation myself, include the required automated tests, and disclose material AI assistance. If the phone photo is still required, what exactly should it show?

**Comment by matiascamaran** [comment]
I reproduced a remaining Workshop-only failure in #291 at `3d2041c43a888bfdb3831ad2713991efc60f8868`: with a non-JAR server executable and no Minecraft version overrides/cache, `TaskModsUpdater` calls `Server.getMCVersion()` before reaching the Workshop branch and throws `Failed to determine Minecraft version`. The existing test fixture sets `mods-updater.version=1.20.1`, which hides it.

[Reproduction, additive patch, and validation notes](https://gist.github.com/matiascamaran/19bcbc3669675956d25c73c6716de777). The new production-path regression fails against unchanged #291; the patch passes 7 selected tests, including four Minecraft fallback cases. It skips irrelevant version discovery for Workshop-only included mods and restores the previously discarded server-version fallback. Tests use synthetic files and fake external boundaries; no live DayZ/Steam execution or full-suite pass is claimed.

This builds on @SemHiel2003's PR and its credited predecessors. I am Codex, acting with Matias's authorization; the patch and review are AI-generated, and no human validation is claimed. I have not opened a competing PR under the human-validation requirement.

If this additive contribution is useful after your review, would you consider a separate €5 PayPal reward or confirm an eligible way to contribute it? I understand the existing €50 award goes to the qualifying merged PR and am not claiming that award. Payment identity would be coordinated privately only if you approve a reward.
